src/retry-strategy/unauthenticated-retry-strategy.ts

```diff
@@ -0,0 +1,49 @@
+import { Logger } from '../logger';
+import { NetworkRequest } from '../request';
+import { BaseSubscription } from '../subscription/base-subscription';
+import { RetryStrategy } from './retry-strategy';
+
+/**
+ * This serves as a no-op implementation of RetryStrategy that just relays whatever it has to the underlying requests. 
+ * Used with ExponentialBackoffRetryStrategy when we don't have a TokenProvider
+ */
+export class UnauthenticatedRetryStrategy implements RetryStrategy {
+    constructor(
+        private logger: Logger
+    ){}
+
+    private subscription: BaseSubscription;
+
+    executeSubscription(
+        error: any,
+        xhrSource: () => XMLHttpRequest, 
+        lastEventId: string,
+        subscriptionCallback: (subscription: BaseSubscription) => void, 
+        errorCallback: (error: any) => void
+    ){
+        let xhr = xhrSource();
+        if(lastEventId){
+            xhr.setRequestHeader("Last-Event-Id", lastEventId);                
+        }
+        this.subscription = new BaseSubscription(
+            xhr, 
+            this.logger, 
+            (headers) => {
+                subscriptionCallback(this.subscription);
+            }, 
+            (error) => {
+                errorCallback(error);
+            } 
+        );
```

Comment:
I think retry strategy should not know about the details of constructing a subscription. Now we're duplicating that logic in all strategies and I'm afraid they will diverge really quickly.